src/app/Instance/BridgeOverview/Components/EventSource.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+  Card,
+  CardTitle,
+  ClipboardCopy,
+  ClipboardCopyVariant,
+  Divider,
+  GridItem,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  Title,
+} from "@patternfly/react-core";
+import React from "react";
+import { DemoData } from "../BridgeOverview";
+import { OBDashboardTableView } from "./OBDashboardTableView";
+import { OBEmptyState } from "./OBEmptyState";
+
+interface EventSourceProps {
+  eventSourceList: DemoData[];
+  onAddingSourceConnector: () => void;
+}
+
+export const EventSource = (props: EventSourceProps): JSX.Element => {
+  const { eventSourceList, onAddingSourceConnector } = props;
+
+  const desc =
+    "Create a source connector to send events from an external system to this bridge";
+
+  return (
+    <>
+      <GridItem lg={4} md={6}>
+        <Card>
+          <CardTitle>Event sources</CardTitle>
+          {eventSourceList.length == 0 ? (
+            <>
+              <OBEmptyState
+                title={"No source connectors"}
+                description={desc}
+                buttonName={"Create source connector"}
+                variant={"secondary"}
+                changeState={onAddingSourceConnector}
+              />
+              <Divider inset={{ default: "insetMd" }} />
+            </>
```

Comment:
About components separation, I see a `CardBody` isn't specified here. 
There's one inside `OBEmptyState` but then the rest of the content (table and ingress endpoint area) are not wrapped inside one. In theory `OBEmptyState` should not know where it's being used but should still work regardless of the nature of its parent (card, grid, flex etc). This will make also eventual future changes easier to apply.

I would place the `CardBody` directly here to wrap all the content below `CardTitle`. You could also move the `Stack` used below and place it as first child of `CardBody` to manage the vertical alignment of the whole component.

I'm suggesting quite some re-organisation here, I know, sorry :sweat_smile: 
We could chat about it before proceeding.